compiler/src/main.cc

```diff
@@ -970,6 +989,12 @@ void parse(t_program* program, t_program* parent_program) {
  * Generate code
  */
 void generate(t_program* program, const vector<string>& generator_strings) {
+  set<std::string>::iterator it = generated_files.find(program->get_path());
```

Comment:
adding code to early abort file generation if the file has already been generated